social_engagement/models.py

```diff
@@ -168,13 +168,37 @@ class StudentSocialEngagementScoreHistory(TimeStampedModel):
     score = models.IntegerField()
 
 
+class StudentSocialEngagementStats(TimeStampedModel):
+    """
+    Stores the social engagement stats for faster access and calculation.
+    """
+    score = models.OneToOneField(
+        StudentSocialEngagementScore,
+        on_delete=models.CASCADE,
+        related_name='stats',
+        primary_key=True
+    )
+    num_threads = models.IntegerField(default=0, db_index=True, null=False)
+    num_thread_followers = models.IntegerField(default=0, db_index=True, null=False)
+    num_replies = models.IntegerField(default=0, db_index=True, null=False)
+    num_flagged = models.IntegerField(default=0, db_index=True, null=False)
+    num_comments = models.IntegerField(default=0, db_index=True, null=False)
+    num_threads_read = models.IntegerField(default=0, db_index=True, null=False)
+    num_downvotes = models.IntegerField(default=0, db_index=True, null=False)
+    num_upvotes = models.IntegerField(default=0, db_index=True, null=False)
+    num_comments_generated = models.IntegerField(default=0, db_index=True, null=False)
```

Comment:
We only need DB indices for these fields if they're going to be present in `WHERE` clauses to avoid full table scans. I looked through the code and didn't find such cases, so we should delete the indexing, otherwise they're going to slow down the performance of new writes because of having to update the index every single time a statistic changes. And I can imagine, because of the large variety of values these fields can take, the data structures for each index will get very large.

If there are plans to do lookups based off of the specific values in these fields somewhere in upcoming code (e.g. in an API), then nevermind this comment.